### ISSUE TITLE
Adding extensions adds the wrong constraints.

### ISSUE THREAD

**Issue by bobdenotter** [bolt/bolt#5456]
If i set up a fresh 2.2.x install, and go to 'extends' to install an extension, for example 'boltforms' or 'labels', I pick the latest `2.x` version. so far, so good:

![screen shot 2016-06-15 at 15 22 02](https://cloud.githubusercontent.com/assets/1833361/16081228/f4303ef2-330c-11e6-9d04-98abc4ad8403.png)

However, if I look in `extensions/composer.json`, I see: 

```
    "require": {
        "bolt/boltforms": "^3.0"
    }
}
```

On the very next extension I install, Bolt will break, because of this: 

![screen shot 2016-06-15 at 15 23 54](https://cloud.githubusercontent.com/assets/1833361/16081301/33bc00a6-330d-11e6-984e-3c56303d0475.png)

**Comment by bobdenotter** [comment]
Perhaps related, on another 2.2.21 site I get the following, when trying to update the extensions: 

```
$ composer instal --no-dev

Loading composer repositories with package information
Updating dependencies
  - Removing bacboslab/menueditor (2.2.1)
  - Installing bacboslab/menueditor (2.2.5)
     Downloading: 100%

> Bolt\Composer\ExtensionInstaller::handle PHP Fatal error:  Call to undefined method Composer\DependencyResolver\Operation\UpdateOperation::getPackage() in /xxx/extensions/ExtensionInstaller.php on line 9
```

**Comment by GawainLynch** [comment]
Unrelated, `extensions/ExtensionInstaller.php` is a hangover from 2.0 that can be safely removed

**Comment by bobdenotter** [comment]
Ok, thanks! 👍

**Comment by GawainLynch** [comment]
Removing the "2.2" aspect form this as it definitely affects 3.x too, it is just that there is nothing production-wise that requires Bolt 4 yet :wink:

**Comment by GawainLynch** [comment]
Fixed in #5473
